Apps/WebClient/test/unit/Services.Test/UserProfileService_Test.cs

```diff
@@ -223,12 +225,44 @@ public void ShouldQueueNotificationUpdate()
                 new Mock<ILegalAgreementDelegate>().Object,
                 cryptoDelegateMock.Object,
                 notificationServiceMock.Object,
-                messageVerificationDelegateMock.Object);
+                messageVerificationDelegateMock.Object,
+                new Mock<IPatientService>().Object);
 
-            RequestResult<UserProfileModel> actualResult = service.CreateUserProfile(new CreateUserRequest() { Profile = userProfile }, new Uri("http://localhost/"), "bearer_token");
+            RequestResult<UserProfileModel> actualResult = await service.CreateUserProfile(new CreateUserRequest() { Profile = userProfile }, new Uri("http://localhost/"), "bearer_token");
             notificationServiceMock.Verify(s => s.QueueNotificationSettings(It.IsAny<NotificationSettingsRequest>()), Times.Once());
             Assert.Equal(ResultType.Success, actualResult.ResultStatus);
             Assert.True(actualResult.ResourcePayload.IsDeepEqual(expected));
         }
+
+        [Fact]
+        public async System.Threading.Tasks.Task ShouldValidateAgeAsync()
+        {
+            string hdid = "1234567890123456789012345678901234567890123456789012";
+            Mock<IConfigurationService> configServiceMock = new Mock<IConfigurationService>();
+            configServiceMock.Setup(s => s.GetConfiguration()).Returns(new ExternalConfiguration() { WebClient = new WebClientConfiguration() { MinPatientAge = 19 } });
+            PatientModel patientModel = new PatientModel() {
+                Birthdate = new DateTime(2010, 1, 1)
```

Comment:
instead of hard-coding the date, can we use a date range indicating number of months from DateTime.Now?
public async System.Threading.Tasks.Task ShouldValidateAgeAsync()([Values(18*12 - 1,18*12 - 2,etc...)] int numberOfMonths) 
... Birthdate = DateTime.Now.Substract(numberOfMonths) // I'm not sure exactly the syntax for this but hopefully you get the ideas.